src/components/guarded-app.tsx

```diff
@@ -0,0 +1,91 @@
+import React, { ReactNode, useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import {
+  AUTH_PARAM,
+  hasRequestExpired,
+  removeRequest,
+  RequestItem,
+  reRunRequest
+} from "../core/guardedRequests.slice";
+import { RootState } from "../core/store";
+import { getCurrentUnixTime } from "../core/utils/helpers/date";
+import {
+  getCurrentLocation,
+  getUrlQueryParam,
+  removeQueryParametersFromUrl,
+  replaceCurrentLocation
+} from "../core/utils/helpers/url";
+import { userIsAnonymous } from "../core/utils/helpers/user";
+import { App } from "../core/utils/types/ids";
+
+export interface GuardedAppProps {
+  app: App;
+  children: ReactNode;
+}
+
+// This component makes sure to withhold app rendering
+// until the persisted request has been executed.
+const GuardedApp = ({ app, children }: GuardedAppProps) => {
+  const dispatch = useDispatch();
+  const { request: persistedRequest } = useSelector(
+    (state: RootState) => state.guardedRequests
+  );
+  const isApplicationBlocked = persistedRequest && !userIsAnonymous();
+  const didAuthenticate = getUrlQueryParam(AUTH_PARAM);
+
+  console.log("PERSISTED REQUEST:", persistedRequest);
```

Comment:
Please silence these errors with an explanation regarding why we think it is necessary to use the `console`.

It would probably make sense to create a function for this as it occurs multiple times in this component.

Also consider using [`console.debug()`](https://developer.mozilla.org/en-US/docs/Web/API/console/debug) to indicate that is in fact something that we intend to keep around for debugging purposes.